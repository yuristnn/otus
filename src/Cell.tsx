import React, { useState } from 'react';

type CellProps = {
    num: number;
};

const Cell: React.FC<CellProps> = ({ num }) => {
    const [isOpen, setIsOpen] = useState(false);

    return (
        <td
            onClick={() => setIsOpen(!isOpen)}
            style={{
                border: '1px solid #000',
                width: 25,
                height: 25,
                textAlign: 'center',
            }}
        >
            {isOpen ? num : ' '}
        </td>
    );
};

export default Cell;
