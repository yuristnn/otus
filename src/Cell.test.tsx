import 'jsdom-global/register';
import React from 'react';
import Cell from './Cell';
import { mount, shallow } from 'enzyme';

describe('Cell component', () => {
    it('Should render Cell component', () => {
        const component = shallow(<Cell num={1} />);
        expect(component).toMatchSnapshot();
    });

    it('Checking cell children and onclick', () => {
        const wrapper = mount(
            <table>
                <tbody>
                    <tr>
                        <Cell num={12} />
                    </tr>
                </tbody>
            </table>
        );
        expect(wrapper.find('td')).toHaveLength(1);
        expect(wrapper.find('td').text()).toContain(' ');
        wrapper.find('td').first().simulate('click');
        expect(wrapper.find('td').text()).toContain(12);
    });
});
