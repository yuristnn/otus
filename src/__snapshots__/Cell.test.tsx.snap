// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component Should render Cell component 1`] = `ShallowWrapper {}`;
